predictsignauxfaibles/__main__.py

```diff
@@ -1,80 +1,207 @@
+    TEST_DATASET.fetch_data()
 
-    train_siren_set = conf.TRAIN_DATASET.data["siren"].unique().tolist()
-    conf.TEST_DATASET.remove_siren(train_siren_set)
+    train_siren_set = TRAIN_DATASET.data["siren"].unique().tolist()
+    TEST_DATASET.remove_siren(train_siren_set)
 
     logging.info(f"{step} - Data preprocessing")
-    conf.TEST_DATASET.replace_missing_data().remove_na(
+    TEST_DATASET.replace_missing_data().remove_na(
         ignore=["time_til_outcome"]
     ).remove_strong_signals()
-    conf.TEST_DATASET.data = run_pipeline(conf.TEST_DATASET.data, conf.TRANSFO_PIPELINE)
-    logging.info(f"{step} - Testing on {len(conf.TEST_DATASET)} observations.")
+    TEST_DATASET.data = run_pipeline(TEST_DATASET.data, conf.TRANSFO_PIPELINE)
+    logging.info(f"{step} - Testing on {len(TEST_DATASET)} observations.")
 
     balanced_accuracy = balanced_accuracy_score(
-        conf.TEST_DATASET.data["outcome"], fit.predict(conf.TEST_DATASET.data)
+        TEST_DATASET.data["outcome"], fit.predict(TEST_DATASET.data)
     )
     logging.info(f"{step} - Test balanced_accuracy: {balanced_accuracy}")
 
     step = "[PREDICT]"
+    args.predict_on = args.predict_on
+    predict_from = args.predict_on + relativedelta(day=1)
+    predict_to = args.predict_on + relativedelta(months=+1)
+    predict_to = predict_to + relativedelta(days=-1)
+
+    PREDICT_DATASET = SFDataset(
+        date_min=predict_from, date_max=predict_to, fields=conf.VARIABLES
+    )
     logging.info(f"{step} - Fetching predict set")
-    conf.PREDICT_DATASET.fetch_data()
+    PREDICT_DATASET.fetch_data()
     logging.info(f"{step} - Data preprocessing")
-    conf.PREDICT_DATASET.replace_missing_data()
-    conf.PREDICT_DATASET.remove_na(ignore=["time_til_outcome", "outcome"])
-    conf.PREDICT_DATASET.data = run_pipeline(
-        conf.PREDICT_DATASET.data, conf.TRANSFO_PIPELINE
-    )
-    logging.info(f"{step} - Predicting on {len(conf.PREDICT_DATASET)} observations.")
-    predictions = fit.predict_proba(conf.PREDICT_DATASET.data)
-    conf.PREDICT_DATASET.data["predicted_probability"] = predictions[:, 1]
+    PREDICT_DATASET.replace_missing_data()
+    PREDICT_DATASET.remove_na(ignore=["time_til_outcome", "outcome"])
+    PREDICT_DATASET.data = run_pipeline(PREDICT_DATASET.data, conf.TRANSFO_PIPELINE)
+    logging.info(f"{step} - Predicting on {len(PREDICT_DATASET)} observations.")
+    predictions = fit.predict_proba(PREDICT_DATASET.data)
+    PREDICT_DATASET.data["predicted_probability"] = predictions[:, 1]
 
     logging.info(f"{step} - Exporting data to csv")
     model_id = datetime.now().strftime("%Y%m%d-%H%M%S")
     export_destination = f"predictions-{model_id}.csv"
-    conf.PREDICT_DATASET.data[["siren", "siret", "predicted_probability"]].to_csv(
+    PREDICT_DATASET.data[["siren", "siret", "predicted_probability"]].to_csv(
         export_destination, index=False
     )
 
 
+parser = argparse.ArgumentParser("main.py", description="Run model prediction")
```

Comment:
I'm a little new to argparse (i usually use `Click` to make CLI) so sorry if that's a dumb question but should it not be `__main__.py` instead of `main.py` ?